sdks/go/pkg/beam/core/runtime/harness/worker_status.go

```diff
@@ -65,20 +70,50 @@ func (w *workerStatusHandler) start(ctx context.Context) error {
 	return nil
 }
 
+func memoryUsage() string {
+	m := runtime.MemStats{}
+	runtime.ReadMemStats(&m)
+	return fmt.Sprintf("\n Total Alloc: %v bytes \n Sys: %v bytes \n Mallocs: %v\n Frees: %v\n HeapAlloc: %v bytes", m.TotalAlloc, m.Sys, m.Mallocs, m.Frees, m.HeapAlloc)
+}
+
+func (w *workerStatusHandler) activeProcessBundleStates() string {
+	var states string
+	for bundleID, store := range w.metStore {
+		execStates := ""
+		for bundleID, state := range store.StateRegistry() {
+			execStates += fmt.Sprintf("ID: %v Execution States: %#v,", bundleID, *state)
+
+		}
+		states += fmt.Sprintf("\nBundle ID: %v\nBundle State: %#v\nBundle Execution States: %v\n", bundleID, *store.BundleState(), execStates)
+	}
+	return states
+}
+
+func (w *workerStatusHandler) cacheStats() string {
+	return fmt.Sprintf("Cache:\n%v", w.cache.CacheMetrics())
+}
+
+func goroutineDump() string {
+	buf := make([]byte, 1<<16)
+	runtime.Stack(buf, true)
```

Comment:
Instead of the raw runtime version, use the pprof version:

`import "runtime/pprof"`
```
profile := pprof.Lookup("goroutine")
if profile != nil {
  // Use debug=1 to get the human readable consolidated goroutine output.
  profile.Write(b, 1)  
}
```

This makes the output easier to read as some repeated goroutines will instead be consolidated, and the duplication indicated with a count.

(Note that a `*strings.Builder` implements `io.Writer` and can be passed right in. (the method is on the pointer)